dockerfile/cargo/Dockerfile

```diff
@@ -130,37 +142,41 @@ FROM busybox:1.34.1-musl AS busybox-full
 FROM rust:1-bullseye as target-arch-libs
 RUN apt update && apt install -y libssl1.1 openssl clang libstdc++6
 
+# Determine shared library dependencies for both bins and libs
 COPY --from=build-env /root/bin /root/bin
+COPY --from=build-env /root/lib/* /root/bin/
 RUN mkdir -p /root/lib_abs && touch /root/lib_abs.list
-RUN bash -c \
-  'i=0; for BIN in /root/bin/*; do \
-    echo "Getting $(uname -m) libs for bin: $BIN"; \
-    readarray -t LIBS < <(ldd "$BIN"); \
-    for LIB in "${LIBS[@]}"; do \
-      PATH1=$(echo $LIB | awk "{print \$1}") ; \
-      if [ "$PATH1" = "linux-vdso.so.1" ]; then continue; fi; \
-      PATH2=$(echo $LIB | awk "{print \$3}") ; \
-      if [ ! -z "$PATH2" ]; then \
-        if cat /root/lib_abs.list | grep -x "$PATH2"; then \
-          echo "Skipping $PATH2, already accounted for"; \
-          continue; \
-        else \
-          echo "Copying lib2: $PATH2"; \
-          cp -L $PATH2 /root/lib_abs/$i ; \
-          echo $PATH2 >> /root/lib_abs.list; \
-        fi; \
-      else \
-        if cat /root/lib_abs.list | grep -x "$PATH1"; then \
-          echo "Skipping $PATH1, already accounted for"; \
-          continue; \
-        else \
-          echo "Copying lib1: $PATH1"; \
-          cp -L $PATH1 /root/lib_abs/$i ; \
-          echo $PATH1 >> /root/lib_abs.list; \
-        fi; \
-      fi; \
-      ((i = i + 1)) ;\
-    done; \
+RUN bash -c 'set -eux;\
+    i=0; for BIN in /root/bin/*; do\
+    echo "Getting $(uname -m) libs for bin: $BIN";\
+    readarray -t LIBS < <(ldd "$BIN");\
+    for LIB in "${LIBS[@]}"; do\
+      PATH1=$(echo $LIB | awk "{print \$1}");\
+      if [ "$PATH1" = "linux-vdso.so.1" ]; then continue; fi;\
+      PATH2=$(echo $LIB | awk "{print \$3}");\
```

Comment:
No change now. But I had a thought. I bet these long bash commands will be easier to read if in a `.sh` or `.bash` file that's copied into the container. We'd get IDE syntax highlighting, for example.

Reply:
Good call. Will handle that as a follow up